uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/instances/leave-request.instance.spec.js

```diff
@@ -248,27 +248,73 @@ define([
-        instance = LeaveRequestInstance.init(helper.createRandomLeaveRequest());
-        promise = instance.calculateBalanceChange();
-      });
+        beforeEach(function () {
+          instance = LeaveRequestInstance.init(helper.createRandomLeaveRequest());
+          promise = instance.calculateBalanceChange();
+        });
 
-      afterEach(function () {
-        $rootScope.$apply();
+        afterEach(function () {
+          $rootScope.$apply();
+        });
+
+        it('calls equivalent API method', function () {
+          promise.then(function () {
+            expect(LeaveRequestAPI.calculateBalanceChange)
+              .toHaveBeenCalledWith(jasmine.objectContaining({
+                contact_id: jasmine.any(String),
+                from_date: jasmine.any(String),
+                to_date: jasmine.any(String),
+                from_date_type: jasmine.any(String),
+                to_date_type: jasmine.any(String)
+              }));
+          });
+        });
       });
 
-      it('calls equivalent API method', function () {
-        promise.then(function () {
-          expect(LeaveRequestAPI.calculateBalanceChange)
-            .toHaveBeenCalledWith(jasmine.objectContaining({
-              contact_id: jasmine.any(String),
-              from_date: jasmine.any(String),
-              to_date: jasmine.any(String),
-              from_date_type: jasmine.any(String),
-              to_date_type: jasmine.any(String)
-            }));
+      describe('when calculation unit is "hours"', function () {
+        var promise;
+
+        beforeEach(function () {
+          instance = LeaveRequestInstance.init(helper.createRandomLeaveRequest());
+          instance.from_date_amount = 0;
+          instance.to_date_amount = 20;
+          promise = instance.calculateBalanceChange('hours');
+        });
+
+        afterEach(function () {
+          $rootScope.$apply();
+        });
+
+        it('calls equivalent API method', function () {
+          promise.then(function () {
+            expect(LeaveRequestAPI.calculateBalanceChange)
+              .toHaveBeenCalledWith(jasmine.objectContaining({
+                contact_id: jasmine.any(String),
+                from_date: jasmine.any(String),
+                to_date: jasmine.any(String)
+              }));
+          });
+        });
+
+        it('amends the breakdown "from" day amount', function () {
+          promise.then(function (balanceChange) {
+            expect(_.first(balanceChange.breakdown).amount).toBe(instance.from_date_amount);
+          });
+        });
+
+        it('amends the breakdown "to" day amount', function () {
+          promise.then(function (balanceChange) {
+            expect(_.last(balanceChange.breakdown).amount).toBe(instance.to_date_amount);
+          });
+        });
+
+        it('amends the breakdown', function () {
+          promise.then(function (balanceChange) {
+            expect(balanceChange.amount).toBe(-23);
```

Comment:
Why a static value like `-23`?

Reply:
As per mocks, I calculated manually looking at mocks and specified that value. If I calculate the value in the test it will look like implementation.